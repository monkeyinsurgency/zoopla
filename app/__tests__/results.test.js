import React from 'react';
import Results from '../pages/results';
import renderer from 'react-test-renderer';

test('Correct postcode returns results', () => {
  const initialProps = {postcode: 'N11'};
  const component = renderer.create(
    <Results qs={initialProps}/>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Incorrect postcode returns no results', () => {
  const initialProps = {postcode: 'W12'};
  const component = renderer.create(
    <Results qs={initialProps}/>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});

test('Page is keyboard accessible', () => {
  const initialProps = {postcode: 'N11'};
  const component = renderer.create(
    <Results qs={initialProps}/>,
  );
  let event = new KeyboardEvent('keypress', {'keyCode': 38});
  document.dispatchEvent(event);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Page is keyboard accessible', () => {
  const initialProps = {postcode: 'N11'};
  const component = renderer.create(
    <Results qs={initialProps}/>,
  );
  let event = new KeyboardEvent('keypress', {'keyCode': 40});
  document.dispatchEvent(event);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

