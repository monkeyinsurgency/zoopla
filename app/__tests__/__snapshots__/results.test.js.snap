// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correct postcode returns results 1`] = `
<div
  className="col-12"
>
  <div
    className="row flex-nowrap justify-content-between align-items-center text-white rounded bg-dark py-3"
  >
    <div
      className="col-12"
    >
      Searching in 
      N11
    </div>
  </div>
  <div
    className="row mb-2"
  >
    <div
      className="col-12"
    >
      <div
        className="results-count"
      >
        5
         properties found
      </div>
      <div>
        <input
          autoFocus={true}
          className="d-none"
        />
        <ul
          className="property-results px-0"
          id="results-list"
        >
          <li
            className="row py-2 property-row active"
            tabIndex={0}
          >
            <div
              className="property-image col-lg-3 col-md-5"
            >
              <img
                src="http://li.zoocdn.com/91062b96be923536c8505e18a132a53f684360d1_354_255.jpg"
              />
            </div>
            <div
              className="property-details col-lg-6 col-md-7"
            >
              <h4>
                5
                 bed 
                Detached house
                 for sale
              </h4>
              <h2>
                <span>
                  £1,800,000
                </span>
              </h2>
              <p
                className="property-copy"
              >
                 Built in 1949 on the site of the former Friern Barnet junior school this substantial five bedroom home sits on a large corner plot occupying approximately one third of an acre. Situated behind secure electronic gates the property which has a south west
              </p>
            </div>
            <div
              className="property-agent col-lg-3 card p-3 col-md-12"
            >
              <div
                className="agent-logo"
              >
                <img
                  src="http://st.zoocdn.com/zoopla_static_agent_logo_(326525).png"
                />
              </div>
              <div
                className="agent-info"
              >
                <ul>
                  <li>
                    Your Move - Spencer & Sener
                  </li>
                  <li>
                    4-6 Station Road, New Barnet
                  </li>
                  <li>
                    EN5 1QW
                  </li>
                  <li>
                    020 3478 3306
                  </li>
                </ul>
              </div>
            </div>
          </li>
          <li
            className="row py-2 property-row "
            tabIndex={1}
          >
            <div
              className="property-image col-lg-3 col-md-5"
            >
              <img
                src="http://li.zoocdn.com/bdf8bcee1c3850bb1ce5c6e7fd3013b50b6bf72d_354_255.jpg"
              />
            </div>
            <div
              className="property-details col-lg-6 col-md-7"
            >
              <h4>
                7
                 bed 
                Flat
                 for sale
              </h4>
              <h2>
                <span>
                  £1,500,000
                </span>
              </h2>
              <p
                className="property-copy"
              >
                Set on the ground floor this large duplex Seven bed two Bath apartment with galleried landing and fully fitted kitchen is offered chain free.
              </p>
            </div>
            <div
              className="property-agent col-lg-3 card p-3 col-md-12"
            >
              <div
                className="agent-logo"
              >
                <img
                  src="http://st.zoocdn.com/zoopla_static_agent_logo_(58080).jpeg"
                />
              </div>
              <div
                className="agent-info"
              >
                <ul>
                  <li>
                    Anscombe & Ringland
                  </li>
                  <li>
                    58-60 Totteridge Lane, Whetstone, London
                  </li>
                  <li>
                    N20 9QP
                  </li>
                  <li>
                    020 3478 3239
                  </li>
                </ul>
              </div>
            </div>
          </li>
          <li
            className="row py-2 property-row "
            tabIndex={2}
          >
            <div
              className="property-image col-lg-3 col-md-5"
            >
              <img
                src="http://li.zoocdn.com/ecb30fde5f377d7f9f3aa7128faab13282bdd73d_354_255.jpg"
              />
            </div>
            <div
              className="property-details col-lg-6 col-md-7"
            >
              <h4>
                4
                 bed 
                Flat
                 for sale
              </h4>
              <h2>
                <span>
                  £1,100,000
                </span>
              </h2>
              <p
                className="property-copy"
              >
                A stunning duplex apartment in prestigious development
              </p>
            </div>
            <div
              className="property-agent col-lg-3 card p-3 col-md-12"
            >
              <div
                className="agent-logo"
              >
                <img
                  src="http://st.zoocdn.com/zoopla_static_agent_logo_(46792).jpeg"
                />
              </div>
              <div
                className="agent-info"
              >
                <ul>
                  <li>
                    Day Morris
                  </li>
                  <li>
                    61 Highgate High Street, Highgate, London
                  </li>
                  <li>
                    N6 5JX
                  </li>
                  <li>
                    020 3478 3348
                  </li>
                </ul>
              </div>
            </div>
          </li>
          <li
            className="row py-2 property-row "
            tabIndex={3}
          >
            <div
              className="property-image col-lg-3 col-md-5"
            >
              <img
                src="http://li.zoocdn.com/e772def64a643bef4939d539ca5450a17b40faf3_354_255.jpg"
              />
            </div>
            <div
              className="property-details col-lg-6 col-md-7"
            >
              <h4>
                4
                 bed 
                Detached house
                 for sale
              </h4>
              <h2>
                <span>
                  £1,000,000
                </span>
              </h2>
              <p
                className="property-copy"
              >
                Situated within good school catchment, close to Bounds Green tube station (Piccadilly Line), a rare opportunity to purchase this immaculately presented detached family home. (contd...)
              </p>
            </div>
            <div
              className="property-agent col-lg-3 card p-3 col-md-12"
            >
              <div
                className="agent-logo"
              >
                <img
                  src="http://st.zoocdn.com/zoopla_static_agent_logo_(326283).png"
                />
              </div>
              <div
                className="agent-info"
              >
                <ul>
                  <li>
                    Bairstow Eves - Muswell Hill Sales
                  </li>
                  <li>
                    36 Muswell Hill Broadway, Muswell Hill
                  </li>
                  <li>
                    N10 3RT
                  </li>
                  <li>
                    020 3463 0154
                  </li>
                </ul>
              </div>
            </div>
          </li>
          <li
            className="row py-2 property-row "
            tabIndex={4}
          >
            <div
              className="property-image col-lg-3 col-md-5"
            >
              <img
                src="http://li.zoocdn.com/47200bfab26d1e67cf23d78d229af60564f13fd3_354_255.jpg"
              />
            </div>
            <div
              className="property-details col-lg-6 col-md-7"
            >
              <h4>
                4
                 bed 
                Semi-detached house
                 for sale
              </h4>
              <h2>
                <span>
                  £975,000
                </span>
              </h2>
              <p
                className="property-copy"
              >
                 Set on a quiet street this fantastic four bedroom semi-detached house offers bright and spacious rooms, fantastic kitchen/dining room, downstairs cloakroom, master bedroom with en suite, and off-street parking. The property is set in a fantastic ...
              </p>
            </div>
            <div
              className="property-agent col-lg-3 card p-3 col-md-12"
            >
              <div
                className="agent-logo"
              >
                <img
                  src="http://st.zoocdn.com/zoopla_static_agent_logo_(161658).png"
                />
              </div>
              <div
                className="agent-info"
              >
                <ul>
                  <li>
                    Foxtons - Barnet
                  </li>
                  <li>
                    98 High Street, High Barnet
                  </li>
                  <li>
                    EN5 5XQ
                  </li>
                  <li>
                    020 7768 0386
                  </li>
                </ul>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`Incorrect postcode returns no results 1`] = `
<div
  className="col-12"
>
  <div
    className="row flex-nowrap justify-content-between align-items-center text-white rounded bg-dark py-3"
  >
    <div
      className="col-12"
    >
      Searching in 
      W12
    </div>
  </div>
  <div
    className="row mb-2"
  >
    <div
      className="col-12"
    >
      <div>
        No results.  Please try a different post code.
      </div>
      <div
        className="col-12"
      >
        <div
          className="row flex-nowrap justify-content-between align-items-center text-white rounded bg-dark py-3"
        >
          <form
            onSubmit={[Function]}
          >
            <label>
              Postcode:
              <input
                autoFocus={true}
                onChange={[Function]}
                type="text"
                value=""
              />
            </label>
            <input
              type="submit"
              value="Submit"
            />
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Page is keyboard accessible 1`] = `
<div
  className="col-12"
>
  <div
    className="row flex-nowrap justify-content-between align-items-center text-white rounded bg-dark py-3"
  >
    <div
      className="col-12"
    >
      Searching in 
      N11
    </div>
  </div>
  <div
    className="row mb-2"
  >
    <div
      className="col-12"
    >
      <div
        className="results-count"
      >
        5
         properties found
      </div>
      <div>
        <input
          autoFocus={true}
          className="d-none"
        />
        <ul
          className="property-results px-0"
          id="results-list"
        >
          <li
            className="row py-2 property-row active"
            tabIndex={0}
          >
            <div
              className="property-image col-lg-3 col-md-5"
            >
              <img
                src="http://li.zoocdn.com/91062b96be923536c8505e18a132a53f684360d1_354_255.jpg"
              />
            </div>
            <div
              className="property-details col-lg-6 col-md-7"
            >
              <h4>
                5
                 bed 
                Detached house
                 for sale
              </h4>
              <h2>
                <span>
                  £1,800,000
                </span>
              </h2>
              <p
                className="property-copy"
              >
                 Built in 1949 on the site of the former Friern Barnet junior school this substantial five bedroom home sits on a large corner plot occupying approximately one third of an acre. Situated behind secure electronic gates the property which has a south west
              </p>
            </div>
            <div
              className="property-agent col-lg-3 card p-3 col-md-12"
            >
              <div
                className="agent-logo"
              >
                <img
                  src="http://st.zoocdn.com/zoopla_static_agent_logo_(326525).png"
                />
              </div>
              <div
                className="agent-info"
              >
                <ul>
                  <li>
                    Your Move - Spencer & Sener
                  </li>
                  <li>
                    4-6 Station Road, New Barnet
                  </li>
                  <li>
                    EN5 1QW
                  </li>
                  <li>
                    020 3478 3306
                  </li>
                </ul>
              </div>
            </div>
          </li>
          <li
            className="row py-2 property-row "
            tabIndex={1}
          >
            <div
              className="property-image col-lg-3 col-md-5"
            >
              <img
                src="http://li.zoocdn.com/bdf8bcee1c3850bb1ce5c6e7fd3013b50b6bf72d_354_255.jpg"
              />
            </div>
            <div
              className="property-details col-lg-6 col-md-7"
            >
              <h4>
                7
                 bed 
                Flat
                 for sale
              </h4>
              <h2>
                <span>
                  £1,500,000
                </span>
              </h2>
              <p
                className="property-copy"
              >
                Set on the ground floor this large duplex Seven bed two Bath apartment with galleried landing and fully fitted kitchen is offered chain free.
              </p>
            </div>
            <div
              className="property-agent col-lg-3 card p-3 col-md-12"
            >
              <div
                className="agent-logo"
              >
                <img
                  src="http://st.zoocdn.com/zoopla_static_agent_logo_(58080).jpeg"
                />
              </div>
              <div
                className="agent-info"
              >
                <ul>
                  <li>
                    Anscombe & Ringland
                  </li>
                  <li>
                    58-60 Totteridge Lane, Whetstone, London
                  </li>
                  <li>
                    N20 9QP
                  </li>
                  <li>
                    020 3478 3239
                  </li>
                </ul>
              </div>
            </div>
          </li>
          <li
            className="row py-2 property-row "
            tabIndex={2}
          >
            <div
              className="property-image col-lg-3 col-md-5"
            >
              <img
                src="http://li.zoocdn.com/ecb30fde5f377d7f9f3aa7128faab13282bdd73d_354_255.jpg"
              />
            </div>
            <div
              className="property-details col-lg-6 col-md-7"
            >
              <h4>
                4
                 bed 
                Flat
                 for sale
              </h4>
              <h2>
                <span>
                  £1,100,000
                </span>
              </h2>
              <p
                className="property-copy"
              >
                A stunning duplex apartment in prestigious development
              </p>
            </div>
            <div
              className="property-agent col-lg-3 card p-3 col-md-12"
            >
              <div
                className="agent-logo"
              >
                <img
                  src="http://st.zoocdn.com/zoopla_static_agent_logo_(46792).jpeg"
                />
              </div>
              <div
                className="agent-info"
              >
                <ul>
                  <li>
                    Day Morris
                  </li>
                  <li>
                    61 Highgate High Street, Highgate, London
                  </li>
                  <li>
                    N6 5JX
                  </li>
                  <li>
                    020 3478 3348
                  </li>
                </ul>
              </div>
            </div>
          </li>
          <li
            className="row py-2 property-row "
            tabIndex={3}
          >
            <div
              className="property-image col-lg-3 col-md-5"
            >
              <img
                src="http://li.zoocdn.com/e772def64a643bef4939d539ca5450a17b40faf3_354_255.jpg"
              />
            </div>
            <div
              className="property-details col-lg-6 col-md-7"
            >
              <h4>
                4
                 bed 
                Detached house
                 for sale
              </h4>
              <h2>
                <span>
                  £1,000,000
                </span>
              </h2>
              <p
                className="property-copy"
              >
                Situated within good school catchment, close to Bounds Green tube station (Piccadilly Line), a rare opportunity to purchase this immaculately presented detached family home. (contd...)
              </p>
            </div>
            <div
              className="property-agent col-lg-3 card p-3 col-md-12"
            >
              <div
                className="agent-logo"
              >
                <img
                  src="http://st.zoocdn.com/zoopla_static_agent_logo_(326283).png"
                />
              </div>
              <div
                className="agent-info"
              >
                <ul>
                  <li>
                    Bairstow Eves - Muswell Hill Sales
                  </li>
                  <li>
                    36 Muswell Hill Broadway, Muswell Hill
                  </li>
                  <li>
                    N10 3RT
                  </li>
                  <li>
                    020 3463 0154
                  </li>
                </ul>
              </div>
            </div>
          </li>
          <li
            className="row py-2 property-row "
            tabIndex={4}
          >
            <div
              className="property-image col-lg-3 col-md-5"
            >
              <img
                src="http://li.zoocdn.com/47200bfab26d1e67cf23d78d229af60564f13fd3_354_255.jpg"
              />
            </div>
            <div
              className="property-details col-lg-6 col-md-7"
            >
              <h4>
                4
                 bed 
                Semi-detached house
                 for sale
              </h4>
              <h2>
                <span>
                  £975,000
                </span>
              </h2>
              <p
                className="property-copy"
              >
                 Set on a quiet street this fantastic four bedroom semi-detached house offers bright and spacious rooms, fantastic kitchen/dining room, downstairs cloakroom, master bedroom with en suite, and off-street parking. The property is set in a fantastic ...
              </p>
            </div>
            <div
              className="property-agent col-lg-3 card p-3 col-md-12"
            >
              <div
                className="agent-logo"
              >
                <img
                  src="http://st.zoocdn.com/zoopla_static_agent_logo_(161658).png"
                />
              </div>
              <div
                className="agent-info"
              >
                <ul>
                  <li>
                    Foxtons - Barnet
                  </li>
                  <li>
                    98 High Street, High Barnet
                  </li>
                  <li>
                    EN5 5XQ
                  </li>
                  <li>
                    020 7768 0386
                  </li>
                </ul>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`Page is keyboard accessible 2`] = `
<div
  className="col-12"
>
  <div
    className="row flex-nowrap justify-content-between align-items-center text-white rounded bg-dark py-3"
  >
    <div
      className="col-12"
    >
      Searching in 
      N11
    </div>
  </div>
  <div
    className="row mb-2"
  >
    <div
      className="col-12"
    >
      <div
        className="results-count"
      >
        5
         properties found
      </div>
      <div>
        <input
          autoFocus={true}
          className="d-none"
        />
        <ul
          className="property-results px-0"
          id="results-list"
        >
          <li
            className="row py-2 property-row "
            tabIndex={0}
          >
            <div
              className="property-image col-lg-3 col-md-5"
            >
              <img
                src="http://li.zoocdn.com/91062b96be923536c8505e18a132a53f684360d1_354_255.jpg"
              />
            </div>
            <div
              className="property-details col-lg-6 col-md-7"
            >
              <h4>
                5
                 bed 
                Detached house
                 for sale
              </h4>
              <h2>
                <span>
                  £1,800,000
                </span>
              </h2>
              <p
                className="property-copy"
              >
                 Built in 1949 on the site of the former Friern Barnet junior school this substantial five bedroom home sits on a large corner plot occupying approximately one third of an acre. Situated behind secure electronic gates the property which has a south west
              </p>
            </div>
            <div
              className="property-agent col-lg-3 card p-3 col-md-12"
            >
              <div
                className="agent-logo"
              >
                <img
                  src="http://st.zoocdn.com/zoopla_static_agent_logo_(326525).png"
                />
              </div>
              <div
                className="agent-info"
              >
                <ul>
                  <li>
                    Your Move - Spencer & Sener
                  </li>
                  <li>
                    4-6 Station Road, New Barnet
                  </li>
                  <li>
                    EN5 1QW
                  </li>
                  <li>
                    020 3478 3306
                  </li>
                </ul>
              </div>
            </div>
          </li>
          <li
            className="row py-2 property-row active"
            tabIndex={1}
          >
            <div
              className="property-image col-lg-3 col-md-5"
            >
              <img
                src="http://li.zoocdn.com/bdf8bcee1c3850bb1ce5c6e7fd3013b50b6bf72d_354_255.jpg"
              />
            </div>
            <div
              className="property-details col-lg-6 col-md-7"
            >
              <h4>
                7
                 bed 
                Flat
                 for sale
              </h4>
              <h2>
                <span>
                  £1,500,000
                </span>
              </h2>
              <p
                className="property-copy"
              >
                Set on the ground floor this large duplex Seven bed two Bath apartment with galleried landing and fully fitted kitchen is offered chain free.
              </p>
            </div>
            <div
              className="property-agent col-lg-3 card p-3 col-md-12"
            >
              <div
                className="agent-logo"
              >
                <img
                  src="http://st.zoocdn.com/zoopla_static_agent_logo_(58080).jpeg"
                />
              </div>
              <div
                className="agent-info"
              >
                <ul>
                  <li>
                    Anscombe & Ringland
                  </li>
                  <li>
                    58-60 Totteridge Lane, Whetstone, London
                  </li>
                  <li>
                    N20 9QP
                  </li>
                  <li>
                    020 3478 3239
                  </li>
                </ul>
              </div>
            </div>
          </li>
          <li
            className="row py-2 property-row "
            tabIndex={2}
          >
            <div
              className="property-image col-lg-3 col-md-5"
            >
              <img
                src="http://li.zoocdn.com/ecb30fde5f377d7f9f3aa7128faab13282bdd73d_354_255.jpg"
              />
            </div>
            <div
              className="property-details col-lg-6 col-md-7"
            >
              <h4>
                4
                 bed 
                Flat
                 for sale
              </h4>
              <h2>
                <span>
                  £1,100,000
                </span>
              </h2>
              <p
                className="property-copy"
              >
                A stunning duplex apartment in prestigious development
              </p>
            </div>
            <div
              className="property-agent col-lg-3 card p-3 col-md-12"
            >
              <div
                className="agent-logo"
              >
                <img
                  src="http://st.zoocdn.com/zoopla_static_agent_logo_(46792).jpeg"
                />
              </div>
              <div
                className="agent-info"
              >
                <ul>
                  <li>
                    Day Morris
                  </li>
                  <li>
                    61 Highgate High Street, Highgate, London
                  </li>
                  <li>
                    N6 5JX
                  </li>
                  <li>
                    020 3478 3348
                  </li>
                </ul>
              </div>
            </div>
          </li>
          <li
            className="row py-2 property-row "
            tabIndex={3}
          >
            <div
              className="property-image col-lg-3 col-md-5"
            >
              <img
                src="http://li.zoocdn.com/e772def64a643bef4939d539ca5450a17b40faf3_354_255.jpg"
              />
            </div>
            <div
              className="property-details col-lg-6 col-md-7"
            >
              <h4>
                4
                 bed 
                Detached house
                 for sale
              </h4>
              <h2>
                <span>
                  £1,000,000
                </span>
              </h2>
              <p
                className="property-copy"
              >
                Situated within good school catchment, close to Bounds Green tube station (Piccadilly Line), a rare opportunity to purchase this immaculately presented detached family home. (contd...)
              </p>
            </div>
            <div
              className="property-agent col-lg-3 card p-3 col-md-12"
            >
              <div
                className="agent-logo"
              >
                <img
                  src="http://st.zoocdn.com/zoopla_static_agent_logo_(326283).png"
                />
              </div>
              <div
                className="agent-info"
              >
                <ul>
                  <li>
                    Bairstow Eves - Muswell Hill Sales
                  </li>
                  <li>
                    36 Muswell Hill Broadway, Muswell Hill
                  </li>
                  <li>
                    N10 3RT
                  </li>
                  <li>
                    020 3463 0154
                  </li>
                </ul>
              </div>
            </div>
          </li>
          <li
            className="row py-2 property-row "
            tabIndex={4}
          >
            <div
              className="property-image col-lg-3 col-md-5"
            >
              <img
                src="http://li.zoocdn.com/47200bfab26d1e67cf23d78d229af60564f13fd3_354_255.jpg"
              />
            </div>
            <div
              className="property-details col-lg-6 col-md-7"
            >
              <h4>
                4
                 bed 
                Semi-detached house
                 for sale
              </h4>
              <h2>
                <span>
                  £975,000
                </span>
              </h2>
              <p
                className="property-copy"
              >
                 Set on a quiet street this fantastic four bedroom semi-detached house offers bright and spacious rooms, fantastic kitchen/dining room, downstairs cloakroom, master bedroom with en suite, and off-street parking. The property is set in a fantastic ...
              </p>
            </div>
            <div
              className="property-agent col-lg-3 card p-3 col-md-12"
            >
              <div
                className="agent-logo"
              >
                <img
                  src="http://st.zoocdn.com/zoopla_static_agent_logo_(161658).png"
                />
              </div>
              <div
                className="agent-info"
              >
                <ul>
                  <li>
                    Foxtons - Barnet
                  </li>
                  <li>
                    98 High Street, High Barnet
                  </li>
                  <li>
                    EN5 5XQ
                  </li>
                  <li>
                    020 7768 0386
                  </li>
                </ul>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
